synfig-core/src/synfig/valuenodes/valuenode_animatedfile.cpp

```diff
@@ -130,21 +168,138 @@ class ValueNode_AnimatedFile::Parser
 							s >> frame >> phoneme;
 							first_frame = min(frame, first_frame);
 							getline(s, word);
-							value[Time(frame*fk)] = phoneme;
+							values[Time(frame*fk)] = phoneme;
 						} else unexpected_end = true;
 					}
-					value[Time(end_frame*fk)] = "rest";
+					values[Time(end_frame*fk)] = "rest";
 				}
 			}
 		}
 		if (first_frame == INT_MAX) first_frame= 1;
-		value[Time((first_frame-1)*fk)] = "rest";
+		values[Time((first_frame-1)*fk)] = "rest";
 
 		if (unexpected_end)
 			warning("Unexpected end of .pgo file. Unsupported format?");
 
 		return true;
 	}
+
+	/// Parse a TSV file (Table Separated Value)
+	/// \param[in] s the TSV stream to be parsed
+	/// \param[out] value the map of time -> layer name
+	/// \param[out] fields the map of metafield -> value
+	/// \return true if successful. There is no guarantee about stream offset in case of failure
+	static bool parse_tsv(istream &s, map<Time, String> &values, map<String, String> &/*fields*/)
+	{
+		try {
+			size_t line_num = 0;
+			std::string line;
+			while (std::getline(s, line)) {
+				line_num++;
+
+				if (line.empty()) {
+					warning(_("TSV file (line %zu): skipping empty line"), line_num);
+					continue;
+				}
+
+				const auto tab_pos = line.find('\t');
+				if (tab_pos == std::string::npos) {
+					error(_("TSV file (line %zu): no TAB separator"), line_num);
+					continue;
+				}
+
+				double seconds = strtodouble(line.substr(0, tab_pos));
+
+				std::string data = line.substr(tab_pos+1);
+
+				Time time(seconds);
+				values[time] = data;
+			}
+		}  catch (...) {
+			return false;
+		}
+
+		return s.eof() || !s.fail();
+	}
+
+	/// Parse a Rhubarb XML file (lipsync software)
+	/// \param[in] s the XML stream to be parsed
+	/// \param[out] value the map of time -> layer name
+	/// \param[out] fields the map of metafield -> value
+	/// \return true if successful. There is no guarantee about stream offset in case of failure
+	static bool parse_xml(istream &s, map<Time, String> &values, map<String, String> &fields)
+	{
+		try {
+			xmlpp::DomParser parser;
+			parser.parse_stream(s);
+			if (parser) {
+				auto doc = parser.get_document();
+				if (!doc) {
+					return false;
+				}
+				auto root = doc->get_root_node();
+				if (!root) {
+					return false;
+				}
+				if (root->get_name() != "rhubarbResult") {
+					synfig::error(_("XML animated file: it only supports Rhubarb Lip Sync files. Found %s"), root->get_name().c_str());
+					return false;
+				}
+				if (auto metadata_node = root->get_first_child("metadata")) {
+					if (auto sound_node = metadata_node->get_first_child("soundFile")) {
+						if (auto sound_element = dynamic_cast<xmlpp::Element*>(sound_node)) {
+							if (auto sound_text_node = sound_element->get_child_text())
+								fields["sound"] = sound_text_node->get_content();
+						}
+						if (fields.count("sound"))
```

Comment:
It looks like it should warn if sound element doesn't load? 
```
if (!fields.count("sound"))
```

Reply:
Whoops!

Reply:
I'll fix the commit and then push --force

Reply:
done!